tools/tsp-client/src/commands.ts

```diff
@@ -485,36 +480,44 @@ export async function generateConfigFilesCommand(argv: any) {
   if (Object.keys(overrideJson).length > 0) {
     emitterPackageJson["overrides"] = overrideJson;
   }
+  
+  const emitterPath = resolveEmitterPathFromArgs(argv) ?? joinPaths(repoRoot, defaultRelativeEmitterPackageJsonPath);
+  
   await writeFile(
-    joinPaths(repoRoot, "eng", "emitter-package.json"),
+    emitterPath,
     JSON.stringify(emitterPackageJson, null, 2),
   );
-  Logger.info(`emitter-package.json file generated in '${joinPaths(repoRoot, "eng")}' directory`);
+  Logger.info(`${basename(emitterPath)} file generated in '${dirname(emitterPath)}' directory`);
 
-  await generateLockFileCommand(argv);
+  await generateLockFileCommandCore(outputDir, emitterPath);
 }
 
 export async function generateLockFileCommand(argv: any) {
-  const outputDir = argv["output-dir"];
-  const repoRoot = await getRepoRoot(outputDir);
+  await generateLockFileCommandCore(
```

Comment:
Same comment as above this command doesnt support the flag you're trying to use...we could enable it if you need this facility.

Reply:
Yes, we will need it. I can update the docs if that works for you?

Reply:
You need to add this flag under the command in the index.ts file and yes please update the docs as well